### PR TITLE
Look up runtime libraries in SDK

### PR DESCRIPTION
**Pull request by beccadax**
In #23175, we started looking in the SDK for swiftmodules, but we want to look for the libraries there too while linking. Fixes <rdar://problem/52059706>.

~~(This PR is not quite fully baked, but I want to kick off Linux tests while I read over the code and check a few details about how this should behave.)~~

This PR ought to be ready to go, but I could use confirmation that it's actually implementing the desired behavior.

### CODE REVIEW

**Comment by brentdax** [comment]
@swift-ci please smoke test

**Comment by brentdax** [comment]
@swift-ci please test

**Review by jrose-apple — APPROVED** [review]

File: lib/Driver/DarwinToolChains.cpp [hunk 1]
```diff
@@ -444,23 +442,34 @@ toolchains::Darwin::constructInvocation(const DynamicLinkJobAction &job,
       }
   }
 
+  bool wantsStaticStdlib =
+      context.Args.hasFlag(options::OPT_static_stdlib,
+                           options::OPT_no_static_stdlib, false);
+
+  SmallVector<std::string, 4> RuntimeLibPaths;
+  getRuntimeLibraryPaths(RuntimeLibPaths, context.Args,
+                         context.OI.SDKPath, /*Shared=*/!wantsStaticStdlib);
```
[the remark this answers is not in the document]
Comment (brentdax) [reply lib/Driver/DarwinToolChains.cpp]: Doing this properly will require a little more work than I want to do right now. I've filed https://bugs.swift.org/browse/SR-11015 to track the idea.

**Comment by brentdax** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by shahmishal** [comment]
@swift-ci please test Windows platform

**Comment by brentdax** [comment]
@swift-ci please smoke test

**Comment by brentdax** [comment]
@swift-ci please test Windows platform

**Pull request merged**